packages/manager/src/queries/regions/regions.ts

```diff
@@ -27,13 +27,24 @@ export const regionQueries = createQueryKeys('regions', {
     },
     queryKey: null,
   },
+  region: (regionId: string) => ({
+    queryFn: () => getRegion(regionId),
+    queryKey: [regionId],
+  }),
   regions: {
     queryFn: getAllRegionsRequest,
     queryKey: null,
   },
 });
 
-export const useRegionsQuery = () =>
+export const useRegionQuery = (regionId: string) => {
+  return useQuery<Region, APIError>({
+    ...regionQueries.region(regionId),
+    enabled: Boolean(regionId),
```

Comment:
For Gecko: I think we'll want to add the `select` here that modifies the `label` so it is consistent with `useRegionsQuery`. 

```suggestion
    enabled: Boolean(regionId),
    select: (region) => ({
      ...region,
      label: getNewRegionLabel(region),
    }),
```